src/cht-script-api-factory.js

```diff
@@ -0,0 +1,61 @@
+const semver = require('semver');
+
+const CoreTargetAggregates = require('./core-target-aggregates');
+
+class ChtScriptApiFactory {
+  constructor(core, pouchdb, appSettings) {
+    this.core = core;
+    this.pouchdb = pouchdb;
+    this.appSettings = appSettings;
+
+    this.coreTargetAggregator = new CoreTargetAggregates(core, pouchdb, appSettings);
+  }
+
+  async getForRulesEngine(defaultUserRoles) {
+    // contact is defined only when using cht api in the contact summary, but should be undefined when calculating rules
+    // facility_id is not used when contact is undefined
+    return this.getForContactSummary(undefined, undefined, undefined, defaultUserRoles);
+  }
+
+  // cht-core/src/ts/services/cht-script-api.service.ts
+  async getForContactSummary(contact, userFacilityId, userContactId, defaultUserRoles) {
+    const datasource = this._getFromDatasource() || this.core.ChtScriptApi;
+    if (!datasource) {
+      throw Error('DataSource and ChtScriptApi are undefined for this core version');
+    }
+
+    const defaultChtPermissionSettings = this.appSettings.permissions;
+    const result = {
+      v1: {
+        hasPermissions: (permissions, userRoles = defaultUserRoles, chtPermissionsSettings = defaultChtPermissionSettings) => {
+          return datasource.v1.hasPermissions(permissions, userRoles, chtPermissionsSettings);
+        },
+        hasAnyPermission: (permissionsGroupList, userRoles = defaultUserRoles, chtPermissionsSettings = defaultChtPermissionSettings) => {
+          return datasource.v1.hasAnyPermission(permissionsGroupList, userRoles, chtPermissionsSettings);
+        },
+      }
+    };
+
+    const userFacilityIds = [userFacilityId];
+    const coreVersion = semver.coerce(this.core.version);
+    if (semver.gte(coreVersion, '4.11.0')) {
```

Comment:
TLDR: I think we should drop support for `4.6` (remove from `dist`) and not include logic like this in the harness code.

I have spent a lot of time going back and forth over this and while it is preferable to support as many versions of the CHT as possible, I think we _MUST_ prioritize minimizing complexity in the harness code or it is going to become unmaintainable.  Same goes for the `core.DataSource` vs `core.ChtScriptApi` logic above.  Technically it is _possible_ to support the alternatives with forking logic, but I just don't think the maintenance burden is worth it.